src/backend/cpu/cholesky.cpp

```diff
@@ -70,13 +71,7 @@ int cholesky_inplace(Array<T> &in, const bool is_upper)
     if(is_upper)
         uplo = 'U';
 
-    int info = 0;
-    auto func = [&] (int& info, Array<T>& in) {
-        info = potrf_func<T>()(AF_LAPACK_COL_MAJOR, uplo, N, in.get(), in.strides()[1]);
-    };
-
-    getQueue().enqueue(func, info, in);
-    getQueue().sync();
+    int info = potrf_func<T>()(AF_LAPACK_COL_MAJOR, uplo, N, in.get(), in.strides()[1]);
```

Comment:
I don't understand what you mean by references. I thought Param were not reference counted so none of that should be going on. I just think its better to be consistant in case there is an issue with thread_local stuff in the future(although its probably unlikely).